yjit_core.c

```diff
@@ -201,6 +201,14 @@ void ctx_set_local_type(ctx_t* ctx, size_t idx, val_type_t type)
 // eg: because of a call we can't track
 void ctx_clear_local_types(ctx_t* ctx)
 {
+    for (int i = 0; i < ctx->stack_size && i < MAX_LOCAL_TYPES; i++) {
```

Comment:
Could you add a one or two sentence comment explaining what this loop does? Besides that LGTM :)